torch/csrc/autograd/input_metadata.h

```diff
@@ -67,6 +68,28 @@ struct InputMetadata {
     return at::zeros(shape_, options_);
   }
 
+  bool is_same_shape(const at::Tensor& grad) const {
+    TORCH_CHECK(!grad.is_nested(), "Nested grads are not currently supported.")
+    return grad.sizes().equals(shape());
+  }
+  bool is_expandable_to_shape(const at::Tensor& grad) const {
+    // TODO: Currently NestedTensors are not expandable.
+    return grad.is_nested() ? false
+                            : at::is_expandable_to(shape(), grad.sizes());
+  }
+
+  std::stringstream incompatible_shape_error_message(
+      const size_t index,
+      const at::Tensor& grad) const {
+    std::stringstream ss;
+    TORCH_CHECK(!grad.is_nested(), "Nested grads are not currently supported.")
```

Comment:
Is this the right macro?

Reply:
TORCH_CHECK bubbles up into a RuntimeError in Python, an internal assert just halts execution. Naturally if it throws an exception in Python the user could catch it etc.